### DISCUSSION TITLE
Updates/Improvement Proposals

### DISCUSSION THREAD

**Discussion by Equious** [Cyfrin/assembly-evm-opcodes-and-formal-verification-course#2] [Ideas]
Hello Everyone!

Cyfrin recognizes the value of radical candor and always encourages the open discussion of constructive feedback and community contribution. Please use this space to bring anything to our attention which may improve the quality of this course and it's content. We welcome things like:

- Error Corrections
- Written Lesson Improvements
- Requests for Clarification or detailing areas which are confusing

If there's a topic you wish was covered in more detail, or you had more information on, mention that here too! Maybe it'll turn into a course of it's own!

Thank you in advance for all your diligence in making Web3 Education better for everyone!

**Comment by Equious** [comment]
In the HorseStore section, while learning about contract creation op codes, I found myself wondering how the RETURN value provided by our transaction was handled by the node operator! I'd be really interested in a brief explainer from the node operators perspective following an `updateNumberOfHorses()` call or something :D

**Comment by Equious** [comment]
We found an issue with the foundry-huff package. When trying to run tests, if a user doesn't have `npm` or `yarn` installed, the `binary_check.sh` script fails with the error `invalid huffc binary`.

A PR was submitted to the Huff team here with a corrected script:  https://github.com/huff-language/foundry-huff/pull/60

**Reply by ie2173** [reply]
Are you using linux or unix?  

I was having issues doing this, and its because binary_check was written in windows.  Pardon me if I'm not 100% correct, but the issue is that since windows/MS-dos creates a new line with "/r/n" while linux uses just "/n", it makes it difficult for linux to understand whats going on.  

What I did was I used a program called dos2unix to convert the files over to be unix compatible and I was able to test with foundry-huff no problem.

**Comment by PatrickAlphaC** [comment]
Looking forward to everyone's comments here!

**Comment by Equious** [comment]
During the Inline Assembly Section of HorseStore we glaze over the testing of the Yul contract and the reasoning for the interface creation really quickly.

**Comment by cromewar** [comment]
Pinning this.
